### ISSUE TITLE
Connection to IRC times out with SASL enabled & No authorization with NickServ by plain messages

### ISSUE THREAD

**Issue by smbdspk** [matrix-org/matrix-appservice-irc#777]
I'm running fully configured plain SASL-enabled inspircd locally, to which clients connect with no problems.
However, this bridge just gets stucked – tested on several machines. And when I'm trying to use PASS instead - it's clearly ignored, both with client and bot accounts. That could be possible because my server doesn't support user auth by server pass, but anyway, what's the problem with SASL?

**Comment by smbdspk** [comment]
Still no positive result with FreeNode and Rizon via registered accounts, ident and pass are fully working there

**Comment by Mikaela** [comment]
Duplicate of https://github.com/matrix-org/matrix-appservice-irc/issues/470 which is blocked on https://github.com/matrix-org/matrix-appservice-irc/issues/673.

If I understand correctly, the SASL support is currently broken as you can configure the password with `!storepass`, but not what username Matrix-appservice-IRC sends to the IRC server.

**Comment by smbdspk** [comment]
Nope, not this. Bridge simply doesn't connect – even with the bot account. Time out, again and again, so the only option is to disable SASL. Username changes just fine, as I can see.

**Comment by Mikaela** [comment]
https://github.com/matrix-org/matrix-appservice-irc/issues/689 then?

**Comment by smbdspk** [comment]
I registered account with matrixirc ident manually from telnet, so it should be accessible initially. The problem is not usernames or somewhat, but the connection itself. 'cause it doesn't fail and doesn't succeed, it doesn't even establish. I guess I should look into packets.
Strange thing is that I had configured bridge from scratch on different machines – and surely missed something

```
ircService:
  servers:
   chat.freenode.net:
      name: "freenode"
      port: 6697
      ssl: true
      sslselfsign: false
      sasl: true
      allowExpiredCerts: false
      # password: 'pa$$w0rd'
      sendConnectionMessages: true
 
     botConfig:
        enabled: true
        nick: "sm5kMatrixBot"
        password: "[censored]"
        joinChannelsIfNoUsers: true
```

**Comment by smbdspk** [comment]
So it DOES connect, but timeouts registration with no response and gets disconnected. Weird.

**Comment by smbdspk** [comment]
Unrelated to bot. I'll try to figure out what's the trouble with inspircd then.

**Comment by smbdspk** [comment]
Got FreeNode working, but Rizon is still broken. Closed, considering that first of all bridge was written for FN.
